eng/common/pipelines/templates/steps/create-pull-request.yml

```diff
@@ -37,6 +37,7 @@ steps:
     }
 
   displayName: Check for changes
+  condition: and(succeeded(), eq(variables['HasChanges'], ''))
```

Comment:
While this change might actually work for our new scenario it will potentially break scenarios people use this template multiple times in the same pipeline. Any usage after the first will always get the result of the first execution of this step which would be incorrect. We will need to make a tweak to how we do this. Instead I think we should create a new template parameter called "SkipCheckingForChanges" which we can use to exclude this step if it is set.